compiler/src/iree/compiler/Dialect/Flow/Transforms/OutlineDispatchRegions.cpp

```diff
@@ -255,7 +255,11 @@ summarizeDispatchWorkgroupsOp(DispatchWorkgroupsOp regionOp) {
       })
       .Case<IREE::LinalgExt::UnsetEncodingOp>([&](auto op) {
         auto opName = getOpNameWithoutDialectName(op);
-        auto encoding = stringifyEnum(op.getSourceTensorEncoding());
+        auto encoding = stringifyEnum(
+            op.getSourceType()
+                .getEncoding()
+                .template cast<IREE::LinalgExt::TensorEncodingAttr>()
```

Comment:
This is not a template method, do we need the `template` keyword here?

Reply:
Clang complained about that. I think it's correct. Notice the `auto op` parameter to the lambda (at line 256). Defining a lambda taking an `auto` parameter makes it implicitly templated. This is a C++14 feature. See https://en.cppreference.com/w/cpp/language/lambda,

> For each parameter in params whose type is specified as auto, an invented template parameter is added to template-params, in order of appearance.

Reply:
I see, I did not check the lambda things, sorry about that and thanks for the reply!